openlr/openlr_simple_decoder.cpp

```diff
@@ -556,49 +615,90 @@ OpenLRSimpleDecoder::OpenLRSimpleDecoder(string const & dataFilename, Index cons
     MYTHROW(DecoderError, ("Can't load file", dataFilename, ":", load_result.description()));
 }
 
-void OpenLRSimpleDecoder::Decode(int const segmentsTohandle, bool const multipointsOnly)
+void OpenLRSimpleDecoder::Decode(string const & outputFilename, int const segmentsToHandle,
+                                 bool const multipointsOnly, int const numThreads)
 {
-  FeaturesRoadGraph roadGraph(m_index, IRoadGraph::Mode::ObeyOnewayTag,
-                              make_unique<CarModelFactory>());
-  RoadInfoGetter roadInfoGetter(m_index);
-  AStarRouter astarRouter(roadGraph, roadInfoGetter);
-  Stats stats;
-
-  ofstream sample("inrix_vs_mwm.txt");
-
   // TODO(mgsergio): Feed segments derectly to the decoder. Parsing sholud not
   // take place inside decoder process.
   vector<LinearSegment> segments;
   if (!ParseOpenlr(m_document, segments))
     MYTHROW(DecoderError, ("Can't parse data."));
 
-  for (auto const & segment : segments)
+  if (multipointsOnly)
   {
-    auto const & ref = segment.m_locationReference;
+    my::EraseIf(segments, [](LinearSegment const & segment) {
+      return segment.m_locationReference.m_points.size() == 2;
+    });
+  }
 
-    if (ref.m_points.size() == 2 && multipointsOnly)
-      continue;
+  if (segmentsToHandle != kHandleAllSegments && segmentsToHandle < segments.size())
+    segments.resize(segmentsToHandle);
+
+  vector<IRoadGraph::TEdgeVector> paths(segments.size());
+
+  size_t constexpr a = LCM(sizeof(LinearSegment), kCacheLineSize) / sizeof(LinearSegment);
```

Comment:
Please add a comment what's going on here and that this is not the most important and valuable part of the program.